### PR TITLE
feat(GDB-10733) Rework and Redesign Cluster Management View

### PR DESCRIPTION
**Pull request by teodossidossev**
## WHAT
-Implemented a new cluster management modal view where all actions related to creating, deleting, and replacing cluster nodes will take place 
- Added a modal confirmation for deleting nodes using ModalService in the deleteNode method.
- Adjusted ClusterContextService methods to handle both Node and Location objects for cluster modifications.
- Refactored the saveNode method to optimize the logic for adding, restoring, or creating nodes in the cluster.
- Introduced handleNodesUpdate to ensure sequential processing of adding and removing nodes.
- Refined cluster management features by adding support for replacing nodes, cancelling node replacements, and node add/delete functionality in cluster configurations
- Updated JSDocs across functions to improve clarity.
- Removed old cluster management code and added new tests.
- Deleted legacy controllers, templates, services, and associated Cypress tests related to the previous cluster management system.
- Added new tests to cover the updated functionality.
- Replaced the handling of node configurations by switching from rpcAddress to endpoint across cluster models and services.

## WHY
Requirements

## HOW
- Refactored URL validation logic in url-utils.js
- Enhanced the validate-url directive to handle protocol exclusions and empty value validations
- Added support for updating cluster groups with a modal dialog in the cluster management UI.
- Improved error handling and state management in cluster-context.service.js and related components.
- Updated cluster-management.controller.js to handle user actions and ensure proper state transitions.
- Applied CSS improvements for the cluster management UI, including styles for autocomplete and error messages.
- Introduced handleNodesUpdate, a new utility function that handles adding/removing nodes in sequence using Promises.
- Added ModalService for confirmation dialogs when deleting nodes and linked it with deleteFromCluster.
- Updated ClusterContextService to use Location.fromJSON for node location creation and enhanced its handling of addition and deletion logic.
- Updated template cluster-list.html to disable node deletion when there are fewer than 3 nodes and added node parameter in deleteNode calls.
- Refactored saveNode method to use a helper function findNodeByEndpoint for cleaner node lookup.
- Introduced replaceNode logic to allow replacing a node with a new one.
- Deleted old Angular controllers, services, templates, and Cypress test files related to the previous cluster management.
- Updated i18n files to remove unnecessary translations.
- Added new Cypress tests and steps to validate the updated cluster management operations.
- Updated the cluster.js model to use endpoint instead of rpcAddress for both adding and removing nodes.

## This MR combines 
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1531
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1536
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1542
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1567
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1564
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1582
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1565
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1553
- https://github.com/Ontotext-AD/graphdb-workbench/pull/1557

### CODE REVIEW

File: src/js/angular/clustermanagement/services/cluster-context.service.js [hunk 1]
```diff
@@ -0,0 +1,308 @@
+import {cloneDeep} from "lodash";
+import {ClusterViewModel} from "../../models/clustermanagement/cluster";
+
+angular
+    .module('graphdb.framework.clustermanagement.services.cluster-context', [])
+    .factory('ClusterContextService', ClusterContextService);
+
+ClusterContextService.$inject = ['EventEmitterService'];
+
+function ClusterContextService(EventEmitterService) {
```
Comment (svilenvelikov) [inline src/js/angular/clustermanagement/services/cluster-context.service.js:10]: Please, verify that the state kept in this service should not be reset at some point, e.g. on page leaving, dialog closing, etc.

Comment (teodossidossev) [reply src/js/angular/clustermanagement/services/cluster-context.service.js:10]: done - moved all the logic in the directive

File: src/js/angular/clustermanagement/controllers/edit-cluster-nodes-modal.controller.js [hunk 2]
```diff
@@ -23,9 +24,25 @@ function EditClusterNodesModalController($scope, $uibModalInstance, $translate,
...
+     * This function is called when the user confirms their actions in the modal.
+     *
+     * It retrieves the current cluster view from `ClusterContextService`.
+     *
+     * If the cluster does not have a valid cluster configuration,
+     * it closes the modal and passes a deep clone of the updated cluster configuration.
+     *
+     * If the cluster does have a valid configuration, it passes a deep clone of the full update actions.
+     *
+     * @private
```
Comment (svilenvelikov) [inline src/js/angular/clustermanagement/controllers/edit-cluster-nodes-modal.controller.js:37]: just to mention here that this is not @private

**Review by svilenvelikov — DISMISSED** [review]

**Comment by svilenvelikov** [comment]
Jenkins retry build

File: src/js/angular/clustermanagement/controllers/edit-cluster-nodes-modal.controller.js (hunk 2 above)
Comment (teodossidossev) [reply src/js/angular/clustermanagement/controllers/edit-cluster-nodes-modal.controller.js:37]: will fix it in the next MR

**Pull request merged**